### PR TITLE
Noop when update without providing any values

### PR DESCRIPTION
**Pull request by billy1624**
## PR Info

- Closes https://github.com/SeaQL/sea-orm/issues/1376

- Dependencies:
  - https://github.com/SeaQL/sea-query/pull/578

## Bug Fixes

- [x] When updating / saving an existing `ActiveModel` with all its value being `Unchanged` or `NotSet`, the expected behaviour should be no-op. That's to skip firing any update statement.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -34,8 +34,8 @@ tracing = { version = "0.1", default-features = false, features = ["attributes",
 rust_decimal = { version = "1", default-features = false, optional = true }
 bigdecimal = { version = "0.3", default-features = false, optional = true }
 sea-orm-macros = { version = "0.10.3", path = "sea-orm-macros", default-features = false, optional = true }
-sea-query = { version = "0.28", features = ["thread-safe"] }
-sea-query-binder = { version = "0.3", default-features = false, optional = true }
+sea-query = { version = "0.28", git = "https://github.com/SeaQL/sea-query", branch = "get-values-from-update-statement", features = ["thread-safe"] }
+sea-query-binder = { version = "0.3", git = "https://github.com/SeaQL/sea-query", branch = "get-values-from-update-statement", default-features = false, optional = true }
```
Comment (tyt2y3) [inline Cargo.toml:38]:
```suggestion
sea-query = { version = "0.28.3", features = ["thread-safe"] }
sea-query-binder = { version = "0.3", default-features = false, optional = true }
```

File: src/executor/update.rs [hunk 2]
```diff
@@ -29,8 +28,9 @@ where
         <A::Entity as EntityTrait>::Model: IntoActiveModel<A>,
         C: ConnectionTrait,
     {
-        // so that self is dropped before entering await
```
Comment (tyt2y3) [inline src/executor/update.rs:32]: :(

**Review by tyt2y3 — APPROVED** [review]

File: src/entity/active_model.rs [hunk 3]
```diff
@@ -1423,8 +1423,8 @@ mod tests {
             vec![
                 Transaction::from_sql_and_values(
                     DbBackend::Postgres,
-                    r#"UPDATE "fruit" SET  WHERE "fruit"."id" = $1 RETURNING "id", "name", "cake_id""#,
```
Comment (tyt2y3) [inline src/entity/active_model.rs:1426]: Didn't know we put such a test case here

Comment (billy1624) [reply src/entity/active_model.rs:1426]: Yep. That's bad :(

Comment (billy1624) [reply src/entity/active_model.rs:1426]: It's a red flag that no one notice

**Pull request merged**